### PR TITLE
@joystream/types - bump to version v0.1.8

### PR DESCRIPTION
**Pull request by mnaamani**
I will publish to npm, based on the latest changes, just stamping it with new version

### CODE REVIEW

**Comment by jfinkhaeuser** [comment]
I'd love to see #199 merged - but I don't know whether that will work with pioneer. What was the rationale for using the beta version here anyway?

**Comment by mnaamani** [comment]
> What was the rationale for using the beta version here anyway?

beta version just came at the point when we pulled from upstream. It wasn't a deliberate choice

**Comment by mnaamani** [comment]
I have no issue bumping types dependencies to newer versions as long as they are compatible with our live chain.

**Comment by jfinkhaeuser** [comment]
Newer is a different issue - I've been using `0.90.1` of all the polkadot packages in the storage node up until now.

**Comment by mnaamani** [comment]
Are you sure 0.90.1 ? The latest seems to be 0.80.1 (released only a few hours ago) and on npm version 0.79.1 is published only.

**Comment by jfinkhaeuser** [comment]
```diff
diff --git a/packages/crypto/package.json b/packages/crypto/package.json
index e7f38ff..9d428db 100644
--- a/packages/crypto/package.json
+++ b/packages/crypto/package.json
@@ -44,9 +44,9 @@
     "temp": "^0.9.0"
   },
   "dependencies": {
-    "@polkadot/keyring": "^0.90.1",
-    "@polkadot/util": "^0.90.1",
-    "@polkadot/util-crypto": "^0.90.1",
+    "@polkadot/keyring": "^0.76.1",
+    "@polkadot/util": "^0.76.1",
+    "@polkadot/util-crypto": "^0.76.1",
     "@polkadot/wasm-crypto": "^0.10.1",
     "sodium": "^3.0.2"
   }@@ -45,12 +44,11 @@
   },
   "dependencies": {
     "@joystream/crypto": "^0.1.0",
-    "@polkadot/api": "^0.78.1",
-    "@polkadot/keyring": "^0.90.1",
-    "@polkadot/ts": "^0.1.56",
-    "@polkadot/types": "^0.78.1",
-    "@polkadot/util-crypto": "^0.90.1",
+    "@joystream/types": "/home/jens/src/joy/apps/packages/joy-types/",
+    "@polkadot/api": "^0.76.1",
+    "@polkadot/keyring": "^0.76.1",
+    "@polkadot/util-crypto": "^0.76.1",
     "lodash": "^4.17.11",
-    "typescript": "^3.4.2"
+    "password-prompt": "^1.1.2"
   }
 }
```

Strange that this mixing of package versions worked. I guess the best thing (for me) is to go with the latest `@polkadot/api`, and the dependencies it has. But maintaining those dependencies manually from `@joystream/types` might be a bit unpleasant... hmm.

**Comment by jfinkhaeuser** [comment]
https://www.npmjs.com/search?q=%40polkadot seems to think that for most packages, `0.76.1` is current, so I'd stick with that IMHO.

**Comment by mnaamani** [comment]
Strictly speaking we only directly depend on @polkadot/types, so to simplify, I updated it to latest published version [0.79.1](https://www.npmjs.com/package/@polkadot/types) to go with matching latest matching version 0.79.1 of the [@polkadot/api](https://www.npmjs.com/package/@polkadot/api)

I don't know why you are seeing a different version on npmjs?

**Comment by mnaamani** [comment]
I'm seeing odd behavior with dependencies when building while the package is inside this repo, vs when moving it outside.

Inside the repo, the dependency on @polkadot/types version isn't being respected. So when bumping dependency to v0.79.1 the build was still against the verison in the top package.json. 
But my depending on the package in an external package, the correct version is used which revealed the error:

`src/members.ts:11:34 - error TS2315: Type 'Enum' is not generic.`

**Comment by mnaamani** [comment]
you can reproduce this by comparing result of:
`yarn && yarn build`
vs
`npm install && npm run build`

**Comment by jfinkhaeuser** [comment]
> I don't know why you are seeing a different version on npmjs?
Looks like `0.79.1` was published three days ago - maybe that?

**Comment by jfinkhaeuser** [comment]
Yarn workspaces are weird, and npm won't understand them. Strange, though.

File: packages/joy-types/package.json [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-    "@polkadot/util-crypto": "^0.76.1",
-    "typescript": "^3.4.5"
+    "@polkadot/types": "=0.77.1"
   },
   "directories": {
     "lib": "lib"
   },
-  "devDependencies": {},
+  "devDependencies": {
+    "typescript": "^3.4.5"
```
Comment (jfinkhaeuser) [inline packages/joy-types/package.json]: Is this going to work with the `postinstall` hook? I'm genuinely curious - on the one hand, absolutely it's a dev dependency. But how does npm handle build tools required during installation?

Comment (mnaamani) [reply packages/joy-types/package.json]: I removed the postinstall hook (it was causing a wierd issue when running yarn in top folder), so I know manually run build and then npm publish, including the compiled code.

**Comment by mnaamani** [comment]
> > I don't know why you are seeing a different version on npmjs?
> 
> Looks like `0.79.1` was published three days ago - maybe that?

That was the version I updated to initially, but 0.79.1 resulted in breaking changes (minor to fix in joy-types (just removing the generic parameters for Enum types, but had to also fix usage in other joy- * packages)

**Comment by mnaamani** [comment]
I tested in example client, with [dependencies](https://github.com/Joystream/joystream-api-examples/blob/master/package.json#L10)

**Comment by jfinkhaeuser** [comment]
The downside of this approach is that I can't use a local development version so easily. But maybe that's something we can resolve later.

**Comment by mnaamani** [comment]
> The downside of this approach is that I can't use a local development version so easily. But maybe that's something we can resolve later.

You mean publishing to npmjs ?

Doesn't `npm link` work for what you need?

**Comment by jfinkhaeuser** [comment]
I can `yarn add` a local path, but since there is no `postinstall` hook, it won't compile anything. It's no big deal, I can do it manually. All I'm saying is that a hook would make this automatic for developers.

But I'm still getting the error with multiple versions... have to dig in now to find where they're coming from on my end.

**Comment by jfinkhaeuser** [comment]
Yup, this seems to do it!

**Comment by jfinkhaeuser** [comment]
One quick suggestion: add a two-liner in the README about how to build & publish this?

**Comment by mnaamani** [comment]
Updated README as suggested, and automated the build step in `prepublish` hook to make sure not to forget.

**Comment by jfinkhaeuser** [comment]
Should close  #91 when this is mered.

**Pull request merged**